diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//arch/x86/include/asm/unistd_32.h ./arch/x86/include/asm/unistd_32.h
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//arch/x86/include/asm/unistd_32.h	2011-03-15 03:20:32.000000000 +0200
+++ ./arch/x86/include/asm/unistd_32.h	2011-07-19 12:59:55.000000000 +0300
@@ -346,10 +346,11 @@
 #define __NR_fanotify_init	338
 #define __NR_fanotify_mark	339
 #define __NR_prlimit64		340
+#define __NR_mysyscall		341
 
 #ifdef __KERNEL__
 
-#define NR_syscalls 341
+#define NR_syscalls 342
 
 #define __ARCH_WANT_IPC_PARSE_VERSION
 #define __ARCH_WANT_OLD_READDIR
diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//arch/x86/kernel/syscall_table_32.S ./arch/x86/kernel/syscall_table_32.S
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//arch/x86/kernel/syscall_table_32.S	2011-03-15 03:20:32.000000000 +0200
+++ ./arch/x86/kernel/syscall_table_32.S	2011-07-19 12:42:28.000000000 +0300
@@ -340,3 +340,4 @@ ENTRY(sys_call_table)
 	.long sys_fanotify_init
 	.long sys_fanotify_mark
 	.long sys_prlimit64		/* 340 */
+	.long sys_mysyscall
diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//include/linux/syscalls.h ./include/linux/syscalls.h
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//include/linux/syscalls.h	2011-03-15 03:20:32.000000000 +0200
+++ ./include/linux/syscalls.h	2011-07-19 13:03:13.000000000 +0300
@@ -832,5 +832,6 @@ asmlinkage long sys_mmap_pgoff(unsigned
 			unsigned long prot, unsigned long flags,
 			unsigned long fd, unsigned long pgoff);
 asmlinkage long sys_old_mmap(struct mmap_arg_struct __user *arg);
+asmlinkage long sys_mysyscall(int i);
 
 #endif
diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//Makefile ./Makefile
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//Makefile	2011-07-18 21:11:02.000000000 +0300
+++ ./Makefile	2011-07-19 12:40:11.000000000 +0300
@@ -491,6 +491,7 @@ drivers-y	:= drivers/ sound/ firmware/
 net-y		:= net/
 libs-y		:= lib/
 core-y		:= usr/
+core-y		+= mysyscall/
 endif # KBUILD_EXTMOD
 
 ifeq ($(dot-config),1)
diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//mysyscall/Makefile ./mysyscall/Makefile
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//mysyscall/Makefile	1970-01-01 03:00:00.000000000 +0300
+++ ./mysyscall/Makefile	2011-07-19 12:42:07.000000000 +0300
@@ -0,0 +1 @@
+obj-y += mysyscall.o
diff -pruNa /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//mysyscall/mysyscall.c ./mysyscall/mysyscall.c
--- /var/tmp/portage/sys-kernel/gentoo-sources-2.6.38-r6/image/usr/src/linux-2.6.38-gentoo-r6//mysyscall/mysyscall.c	1970-01-01 03:00:00.000000000 +0300
+++ ./mysyscall/mysyscall.c	2011-07-19 12:42:07.000000000 +0300
@@ -0,0 +1,7 @@
+#include <linux/linkage.h>
+#include <linux/fs.h>
+
+asmlinkage long sys_mysyscall(int i)
+{
+	return (long) i+10;
+}
